src/lib/logger.ts

```diff
@@ -0,0 +1,364 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { Writable } from 'stream';
+import util from 'util';
+import winston from 'winston';
+import Transport from 'winston-transport';
+
+interface VIPLoggerOptions {
+	logDir?: string;
+	consoleLevel?: string;
+	fileLevel?: string;
+}
+
+interface DebugLikeLogger {
+	( ...args: unknown[] ): boolean;
+	enabled: boolean;
+	namespace: string;
+	error: ( ...args: unknown[] ) => boolean;
+	warn: ( ...args: unknown[] ) => boolean;
+	info: ( ...args: unknown[] ) => boolean;
+	verbose: ( ...args: unknown[] ) => boolean;
+	debug: ( ...args: unknown[] ) => boolean;
+	silly: ( ...args: unknown[] ) => boolean;
+}
+
+class VIPLogger {
+	private logDir: string;
+	private rootLogger: winston.Logger;
+	private loggers: Map< string, DebugLikeLogger >;
+	private enabledNamespaces: Set< string >;
+	private isDebugMode: boolean;
+	private commandName: string;
+	private logFilePath: string;
```

Comment:
```suggestion
	private readonly logDir: string;
	private readonly rootLogger: winston.Logger;
	private readonly loggers: Map< string, DebugLikeLogger >;
	private readonly enabledNamespaces: Set< string >;
	private readonly isDebugMode: boolean;
	private readonly commandName: string;
	private readonly logFilePath: string;
```

They are never reassigned.